// @vercel/sdk/esm/models/removeprojectdomainop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const RemoveProjectDomainRequest$inboundSchema = z.object({
    idOrName: z.string(),
    domain: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
/** @internal */
export const RemoveProjectDomainRequest$outboundSchema = z.object({
    idOrName: z.string(),
    domain: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var RemoveProjectDomainRequest$;
(function (RemoveProjectDomainRequest$) {
    /** @deprecated use `RemoveProjectDomainRequest$inboundSchema` instead. */
    RemoveProjectDomainRequest$.inboundSchema = RemoveProjectDomainRequest$inboundSchema;
    /** @deprecated use `RemoveProjectDomainRequest$outboundSchema` instead. */
    RemoveProjectDomainRequest$.outboundSchema = RemoveProjectDomainRequest$outboundSchema;
})(RemoveProjectDomainRequest$ || (RemoveProjectDomainRequest$ = {}));
export function removeProjectDomainRequestToJSON(removeProjectDomainRequest) {
    return JSON.stringify(RemoveProjectDomainRequest$outboundSchema.parse(removeProjectDomainRequest));
}
export function removeProjectDomainRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectDomainRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectDomainRequest' from JSON`);
}
/** @internal */
export const RemoveProjectDomainResponseBody$inboundSchema = z.object({});
/** @internal */
export const RemoveProjectDomainResponseBody$outboundSchema = z.object({});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var RemoveProjectDomainResponseBody$;
(function (RemoveProjectDomainResponseBody$) {
    /** @deprecated use `RemoveProjectDomainResponseBody$inboundSchema` instead. */
    RemoveProjectDomainResponseBody$.inboundSchema = RemoveProjectDomainResponseBody$inboundSchema;
    /** @deprecated use `RemoveProjectDomainResponseBody$outboundSchema` instead. */
    RemoveProjectDomainResponseBody$.outboundSchema = RemoveProjectDomainResponseBody$outboundSchema;
})(RemoveProjectDomainResponseBody$ || (RemoveProjectDomainResponseBody$ = {}));
export function removeProjectDomainResponseBodyToJSON(removeProjectDomainResponseBody) {
    return JSON.stringify(RemoveProjectDomainResponseBody$outboundSchema.parse(removeProjectDomainResponseBody));
}
export function removeProjectDomainResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectDomainResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectDomainResponseBody' from JSON`);
}
